v1/image.go

```diff
@@ -55,4 +58,29 @@ type Image interface {
 	// LayerByDiffID is an analog to LayerByDigest, looking up by "diff id"
 	// (the uncompressed hash).
 	LayerByDiffID(Hash) (Layer, error)
+
+	//////////////////////////////////////////////////////
+	// All method below are for setting functional options
+	//////////////////////////////////////////////////////
+
+	// SetTransport allows a transport configuration to be passed down
+	// to the underlying implementation in the image.
+	SetTransport(http.RoundTripper) error
+
+	// SetAuth allows auth configuration to be passed down to the
+	// underlying http.Client.
+	SetAuth(authn.Keychain) error
```

Comment:
I am 👎 on `s/Authenticator/Keychain/`.  Folks shouldn't have to wrap an `Authenticator` in a `Keychain` to use them.

Reply:
That's fair. The main reason I did this is to get rid of the auth boilerplate that's hanging around everywhere, but it does make things less convenient for callers.

Reply:
WDYT about making keychains implement `Authenticator` so that you can use them interchangeably?